server/bot/commands/trivia.ts

```diff
@@ -0,0 +1,259 @@
+    usage: '/trivia <topic> (--bot-plays=true|false) (--time-limit=<time longer than 4s>) (--max-score=<points greater than 0>) (--reveal-answer=true|false)\n During a trivia session, the following commands may also be run:\n/trivia score\n/trivia stop',
+    example: '/trivia greekmyth --bot-plays=true',
+    guildOnly: true,
+    argsRequired: true
+  },
+  generator: async (message: Message, args: string[], { tempDB, client }) => {
+    let botPlays = false
+    let timeLimit = 15000
+    let maxScore = 30
+    let revealAnswer = true
+
+    if (args.find(element => element.includes('--bot-plays='))) {
+      if (args.find(element => element.includes('--bot-plays=')).split('=')[1] === 'true') {
+        botPlays = true
+      } else if (args.find(element => element === '--bot-plays=').split('=')[1] !== 'false') {
+        return 'Invalid usage. It must be either true or false.'
+      }
+    }
+    if (args.find(element => element.includes('--time-limit='))) {
+      if (+args.find(element => element.includes('--time-limit=')).split('=')[1] > 4) {
+        timeLimit = +args.find(element => element.includes('--time-limit=')).split('=')[1]
+      } else {
+        return 'Invalid usage. It must be a number greater than 4.'
+      }
+    }
+    if (args.find(element => element.includes('--max-score='))) {
+      if (+args.find(element => element.includes('--max-score=')).split('=')[1] > 0) {
+        maxScore = +args.find(element => element.includes('--max-score')).split('=')[1]
+      } else {
+        return 'Invalid usage. It must be a number greater than 0.'
+      }
+    }
+    if (args.find(element => element.includes('--reveal-answer='))) {
+      if (args.find(element => element.includes('--reveal-answer=')).split('=')[1] === 'false') {
+        revealAnswer = false
+      } else if (args.find(element => element === '--reveal-answer=').split('=')[1] !== 'true') {
+        return 'Invalid usage. It must be either true or false.'
+      }
+    }
+    if (args.length === 1 && ['scores', 'score', 'leaderboard'].includes(args[0])) {
+      const session = tempDB.trivia[message.channel.id]
+      if (session) {
+        return session.getScores()
+      } else {
+        return 'There is no trivia session ongoing in this channel.'
+      }
+    } else if (args.length === 1 && args[0] === 'list') {
+      let lists = await fs.promises.readdir('./triviaLists/')
+      lists = lists.map(list => list.slice(0, -4))
+      const member = message.member.guild.members.get(client.user.id)
+      const color = member ? (member.roles.map(i => member.guild.roles.get(i)).sort(
+        (a, b) => a.position > b.position ? -1 : 1
+      ).find(i => i.color !== 0) || { color: 0 }).color : 0
+      const embed: EmbedOptions = {
+        title: 'Available trivia lists',
+        color,
+        fields: lists.map(name => ({ name, value: '_ _', inline: true }))
```

Comment:
I think this is a little better and allows for non-txt extension files.
```suggestion
      const lists = await fs.promises.readdir('./triviaLists/')
      const member = message.member.guild.members.get(client.user.id)
      const color = member ? (member.roles.map(i => member.guild.roles.get(i)).sort(
        (a, b) => a.position > b.position ? -1 : 1
      ).find(i => i.color !== 0) || { color: 0 }).color : 0
      const embed: EmbedOptions = {
        title: 'Available trivia lists',
        color,
        fields: lists.map(name => ({ name: name.replace('.txt', ''), value: '_ _', inline: true }))
```